modules/peridynamics/src/base/PeridynamicsApp.C

```diff
@@ -0,0 +1,101 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PeridynamicsApp.h"
+#include "TensorMechanicsApp.h" // tensor mechanics dependency
+#include "Moose.h"
+#include "AppFactory.h"
+#include "MooseSyntax.h"
+
+template <>
+InputParameters
+validParams<PeridynamicsApp>()
+{
+  InputParameters params = validParams<MooseApp>();
+  return params;
+}
+
+registerKnownLabel("PeridynamicsApp");
+
+PeridynamicsApp::PeridynamicsApp(const InputParameters & parameters) : MooseApp(parameters)
+{
+  PeridynamicsApp::registerAll(_factory, _action_factory, _syntax);
+}
+
+PeridynamicsApp::~PeridynamicsApp() {}
+
+void
+PeridynamicsApp::registerApps()
+{
+  registerApp(PeridynamicsApp);
+}
+
+void
+PeridynamicsApp::registerObjects(Factory & factory)
+{
+  mooseDeprecated("use registerAll instead of registerObjects");
+  Registry::registerObjectsTo(factory, {"PeridynamicsApp"});
+}
+
+static void
+associateSyntaxInner(Syntax & syntax, ActionFactory & /*action_factory*/)
+{
```

Comment:
Let's modify this to have this follow the convention of the TensorMechanics master, so that in the input file you have something like this:
```
[Modules/Peridynamics/Master]
  [my_name]
    stuff
  []
[]